### PR TITLE
[benchmark] Create files once

### PR DESCRIPTION
**Pull request by janlemcke**
Until now, we always created and deleted the benchmark file that we want to read from for each benkmark. It's finally time to change this!

### CODE REVIEW

File: src/benchmark/file_io_read_micro_benchmark.hpp [hunk 1]
```diff
@@ -10,28 +12,32 @@ namespace hyrise {
 class FileIOMicroReadBenchmarkFixture : public MicroBenchmarkBasicFixture {
  public:
   void SetUp(::benchmark::State& state) override {
-    NUMBER_OF_BYTES = _align_to_pagesize(state.range(0));
+    auto size_parameter = state.range(0);
```
Comment (tjjordan) [inline src/benchmark/file_io_read_micro_benchmark.hpp:15]: const

File: src/benchmark/file_io_read_micro_benchmark.hpp [hunk 2]
```diff
@@ -10,28 +12,32 @@ namespace hyrise {
...
+    NUMBER_OF_BYTES = _align_to_pagesize(size_parameter);
     NUMBER_OF_ELEMENTS = NUMBER_OF_BYTES / uint32_t_size;
-
-    // each int32_t contains four bytes
-    numbers = generate_random_positive_numbers(NUMBER_OF_ELEMENTS);
-    control_sum = std::accumulate(numbers.begin(), numbers.end(), uint64_t{0});
+    filename = "benchmark_file_" + std::to_string(size_parameter) + ".txt";
```
Comment (tjjordan) [inline src/benchmark/file_io_read_micro_benchmark.hpp:18]: "benchmark_data" would be a little bit more meaningful.

File: src/benchmark/file_io_read_micro_benchmark.hpp [hunk 3]
```diff
@@ -10,28 +12,32 @@ namespace hyrise {
...
-    Assert(((fd = creat(filename, O_WRONLY)) >= 1), close_file_and_return_error_message(fd, "Create error: ", errno));
-    chmod(filename, S_IRWXU);  // enables owner to rwx file
-    Assert((write(fd, std::data(numbers), NUMBER_OF_BYTES) == NUMBER_OF_BYTES),
-           close_file_and_return_error_message(fd, "Write error: ", errno));
-    close(fd);
-  }
-
-  void TearDown(::benchmark::State& /*state*/) override {
-    Assert(std::remove(filename) == 0, "Remove error: " + std::strerror(errno));
+    if (!std::filesystem::exists(filename.c_str())) {
```
Comment (tjjordan) [inline src/benchmark/file_io_read_micro_benchmark.hpp:21]: Does a C++ library function really need a c string as input?

File: src/benchmark/file_io_read_micro_benchmark.hpp [hunk 4]
```diff
@@ -10,28 +12,32 @@ namespace hyrise {
...
+        numbers = generate_random_positive_numbers(NUMBER_OF_ELEMENTS);
+        control_sum = std::accumulate(numbers.begin(), numbers.end(), uint64_t{0});
+        Assert(((fd = creat(filename.c_str(), O_WRONLY)) >= 1), close_file_and_return_error_message(fd, "Create error: ", errno));
+        chmod(filename.c_str(), S_IRWXU);  // enables owner to rwx file
+        Assert((write(fd, std::data(numbers), NUMBER_OF_BYTES) == NUMBER_OF_BYTES),
+               close_file_and_return_error_message(fd, "Write error: ", errno));
+        close(fd);
+    }else{
```
Comment (tjjordan) [inline src/benchmark/file_io_read_micro_benchmark.hpp:29]:
```suggestion
    } else {
```

File: src/benchmark/file_io_read_micro_benchmark.hpp [hunk 5]
```diff
@@ -10,28 +12,32 @@ namespace hyrise {
...
+        Assert(((fd = open(filename.c_str(), O_RDONLY)) >= 0), close_file_and_return_error_message(fd, "Open error: ", errno));
+        auto* map = reinterpret_cast<int32_t*>(mmap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, 0));
+        Assert((map != MAP_FAILED), close_file_and_return_error_message(fd, "Mapping Failed: ", errno));
+        auto map_span_view = std::span{map, NUMBER_OF_ELEMENTS};
```
Comment (tjjordan) [inline src/benchmark/file_io_read_micro_benchmark.hpp:33]: nice!

Comment (tjjordan) [inline src/benchmark/file_io_read_micro_benchmark.hpp:33]: const

**Review by tjjordan — APPROVED** [review]
Lgtm! 👍 
Please consider running the lint and the format script before commiting.

**Comment by rrcomtech** [comment]
I looked at the dir while running the benchmark and can confirm, that there is no file created or deleted.
<details>
<summary>Watchdog Script</summar>

```python
import sys
import time
import logging
from watchdog.observers import Observer
from watchdog.events import LoggingEventHandler

if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO,
                        format='%(asctime)s - %(message)s',
                        datefmt='%Y-%m-%d %H:%M:%S')
    path = sys.argv[1] if len(sys.argv) > 1 else '.'
    event_handler = LoggingEventHandler()
    observer = Observer()
    observer.schedule(event_handler, path, recursive=True)
    observer.start()
    try:
        while True:
            time.sleep(1)
    except KeyboardInterrupt:
        observer.stop()
    observer.join()
```
</details>

**Review by tjjordan — APPROVED** [review]

**Pull request merged**